File: src/main/java/org/featx/jvmon/entity/AbstractAudit.java
package org.featx.jvmon.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Excepts
 * @since 2019/10/15 0:34
 */

@Data
@EqualsAndHashCode(callSuper = true)
abstract class AbstractAudit<I> extends AbstractUpdate<I> implements Audit<I> {

    private String createdBy;

    private String updatedBy;

}

File: src/main/kotlin/org/featx/jvmon/entity/AbstractAudit.kt
package org.featx.jvmon.entity

import lombok.Data
import lombok.EqualsAndHashCode

@Data
@EqualsAndHashCode(callSuper = true)
abstract class AbstractAudit<I> : AbstractUpdate<I>(), Audit<I> {

    override var createdBy: String? = null
        set(id) {
            field = this.createdBy
        }

    override var updatedBy: String? = null
        set(deleted) {
            field = this.updatedBy
        }
}
